src/vanilla.ts

```diff
@@ -107,23 +107,50 @@ const buildProxyFunction = (
       : Object.create(Object.getPrototypeOf(target))
     markToTrack(snap, true) // mark to track
     snapCache.set(target, [version, snap])
-    Reflect.ownKeys(target).forEach((key) => {
+
+    function getValue(key: PropertyKey): any {
```

Comment:
I needed to use `Object.defineProperty` instead of `snap[key] = value` (see in-source comment later), so extracted the current "is the value a ref or proxy" behavior into a function.